packages/block-library/src/spacer/edit.js

```diff
@@ -197,7 +254,51 @@ const SpacerEdit = ( {
 				width: '72px',
 			} );
 		}
-	}, [] );
+		if (
+			isFlexLayout &&
+			selfStretch !== 'fill' &&
+			selfStretch !== 'fit' &&
+			! flexSize
+		) {
+			const newSize =
+				inheritedOrientation === 'horizontal'
+					? getSpacingPresetCssVar( width ) || '72px'
+					: getSpacingPresetCssVar( height ) || '100px';
```

Comment:
This seems to be working a bit better now, but it looks like there's still a couple of issues when dragging a Spacer block in and out of a flex container. Here's a quick gif:

![2023-04-13 15 18 19](https://user-images.githubusercontent.com/14988353/231661693-5688a696-4840-44a0-8096-0c818b24ff2d.gif)

I'm not too sure what's happening, unfortunately I've found it a little hard to figure out, but my observations so far are:

* The conversion from the spacing preset to the `px` value doesn't appear to populate the Fixed input field
* When moving into the flex container block and out again, somewhere along the way, it seems that the size values become stale. So some of the time the value appears to be inconsistent (like it's remembering the one set in the flex state or vice-versa)

Reply:
> The conversion from the spacing preset to the px value doesn't appear to populate the Fixed input field

It's not converting, so the input doesn't understand the value of the preset. Perhaps we should make it possible to set preset sizes in the child layout controls; I'm sure it would be useful outside of this scenario?

> When moving into the flex container block and out again, somewhere along the way, it seems that the size values become stale

Oooh well spotted! We're alternately setting `height` and `flexSize` and not changing either of them when setting the other, so the block attributes end up something like: `{"height":"var:preset|spacing|40","style":{"layout":{"flexSize":"130px","selfStretch":"fixed"}}}`. I'll look into it!

Reply:
> I wonder if we'd ultimately want a different scale for size presets than spacing, given that width and height values will often need to be much bigger than spacing between things?

Yeah, that's a great point! I'll convert to px for now.